lib/rubocop/cop/rails/migration_timestamp.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+require 'time'
+
+module RuboCop
+  module Cop
+    module Rails
+      # Checks that migration file names start with a valid timestamp.
+      #
+      # @example
+      #   # bad
+      #   # db/migrate/bad.rb
+
+      #   # bad
+      #   # db/migrate/123_bad.rb
+
+      #   # bad
+      #   # db/migrate/20171301000000_bad.rb
+      #
+      #   # good
+      #   # db/migrate/20170101000000_good.rb
+      #
+      class MigrationTimestamp < Base
+        include RangeHelp
+
+        MSG = 'Migration file name must start with a valid `YYYYmmddHHMMSS_` timestamp in the past.'
+
+        def on_new_investigation
+          file_path = processed_source.file_path
+          return unless file_path.include?('db/migrate')
+
+          timestamp = File.basename(file_path).split('_', 2).first
+          return if valid_timestamp?(timestamp)
+
+          add_offense(source_range(processed_source.buffer, 1, 0))
+        end
+
+        private
+
+        def valid_timestamp?(timestamp)
+          format =                    '%Y%m%d%H%M%S'
+          format_with_utc_suffix =    '%Y%m%d%H%M%S %Z'
```

Comment:
I'm getting real nitpicky, but just in case:

```suggestion
          format_with_utc_suffix =    '#{format} %Z'
```

Reply:
I think I'd kept the literal so the formatting would line up, and so we don't create a string instance every time (literals with interpolation aren't frozen), but yeah, it probably makes more sense to interpolate (and this isn't a hot path anyway).

👍 I've moved `format` to be a keyword argument, and switched to interpolating it in to this variable.